src/pages/index.js

```diff
@@ -0,0 +1,392 @@
+});
+
+avatarAddButton.addEventListener("click", () => {
+  avatarLinkInput.value = avatarImage.textContent;
+  resetValidation(avatarForm, [avatarLinkInput], settings);
+  openModal(avatarModal);
+});
+
+deleteCancelBtn.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+avatarCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+profileCloseBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+cardAddButton.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+cardCloseButton.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+previewModalCloseBtn.addEventListener("click", () => {
+  closeModal(cardPreviewModal);
+});
+
+modalOverlays.forEach((modal) => {
+  modal.addEventListener("click", (event) => {
+    // Check if the click happened on the overlay, not on the content inside the modal
+    if (event.target === modal) {
+      closeModal(modal);
+    }
+  });
+});
+
+function closeEscapeBtn(evt) {
+  modalOverlays.forEach((modal) => {
+    if (evt.key === "Escape") {
+      closeModal(modal);
+    }
+  });
+}
+
+// modalOverlay.forEach((modal) => {
+//   modal.addEventListener("keydown", (event) => {
+//     // Check if the click happened on the overlay, not on the content inside the modal
+//     if (event.key === "Escape") {
+//       closeModal(modal);
+//     }
+//   });
+// });
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeEscapeBtn);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  // submitBtn.textContent = "Saving..."
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      // todo - use data argument instead of value
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // TODO - call setButtonText instead
+      // submitBtn.textContent = "Save";
+      setButtonText(submitBtn, false);
+    });
+}
+// TODO implement loading text for call form submissions
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  setButtonText(cardSubmitButton, true);
+
+  api
+    .postUserCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      closeModal(cardModal);
+      //   name: cardNameInput.value,
+      //   link: cardLinkInput.value,
+    })
+    // const cardElement = getCardElement(inputValues);
+    // cardsList.prepend(cardElement);
+    // evt.target.reset();
+    // disableButton(cardSubmitButton, settings);
```

Comment:
Please, use it in line 345 for disabling

* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.